backend/routes/search.ts

```
@@ -0,0 +1,104 @@
+
+const router = Router();
+
+/**
+ * @route   GET /api/search/vector
+ * @desc    Performs vector search based on query text.
+ * @access  Public (for now, consider auth later if needed)
+ * @query   {string} query - The search text.
+ * @returns {VectorSearchResponse} Search results with score and original data.
+ */
+router.get<
+    {},
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)